src/dotnet/Telemetry/ExternalTelemetryProperties.cs

```diff
@@ -0,0 +1,125 @@
+        /// For Unix, or on error, currently returns empty string.
+        /// </summary>
+        internal static string GetInstallationType()
+        {
+            if (RuntimeEnvironment.OperatingSystemPlatform != Platform.Windows)
+                return "";
+
+            const string Key = @"HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion";
+            const string ValueName = @"InstallationType";
+
+            try
+            {
+                return (string)Registry.GetValue(Key, ValueName, defaultValue: "");
+            }
+            // Catch everything: this is for telemetry only.
+            catch (Exception e)
+            {
+                Debug.Assert(e is ArgumentException | e is SecurityException | e is InvalidCastException);
+                return "";
+            }
+        }
+
+        [DllImport("kernel32.dll", ExactSpelling = true, SetLastError = false)]
+        private static extern bool GetProductInfo(uint dwOSMajorVersion, uint dwOSMinorVersion, uint dwSpMajorVersion, uint dwSpMinorVersion, out uint pdwReturnedProductType);
+
+        /// <summary>
+        /// For Windows, returns the product type, loosely the SKU, as encoded by GetProductInfo().
+        /// For example, Enterprise is "4" (0x4) and Professional is "48" (0x30)
+        /// See https://msdn.microsoft.com/en-us/library/windows/desktop/ms724358(v=vs.85).aspx for the full list.
+        /// We're not attempting to decode the value on the client side as new Windows releases may add new values.
+        /// For Unix, or on error, returns an empty string.
+        /// </summary>
+        /// <returns></returns>
+        internal static string GetProductType()
+        {
+            if (RuntimeEnvironment.OperatingSystemPlatform != Platform.Windows)
+                return "";
+
+            try
+            {
+                if (GetProductInfo((uint)Environment.OSVersion.Version.Major, (uint)Environment.OSVersion.Version.Minor, 0, 0, out uint productType))
+                {
+                    return productType.ToString("D");
+                }
+            }
+            // Catch everything: this is for telemetry only
+            catch(Exception e)
+            {
+                Debug.Assert(false, $"Unexpected exception from GetProductInfo: ${e.GetType().Name}: ${e.Message}");
+            }
+
+            return "";
+        }
+
+        [DllImport("libc", ExactSpelling = true, CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
+        private static extern IntPtr gnu_get_libc_release();
+
+        [DllImport("libc", ExactSpelling = true, CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
+        private static extern IntPtr gnu_get_libc_version();
+
+        /// <summary>
+        /// If gnulibc is available, returns the release, such as "stable".
+        /// If the libc is musl, currently returns empty string.
```

Comment:
Is this not possible?  or is it just a lower priority?

Reply:
Dug through Musl briefly and did not find a equivalent. As for version it does not seem to be exposed by an API.